### PR TITLE
🏠 Repo & Build system setup

### PR DESCRIPTION
**Pull request by jakubfiala**
I'm not too happy that the package-lock.json file is 7k lines, but npm says it should be commited. I'm ready to remove it if we don't want it there.

Anyway, this PR:

- sets up the npm package
- creates some basic folder structure
- integrates webpack for both JS and SCSS
- uses nodemon for file change watching in development

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This is the directory where the static site gets built.
```
Comment (jamesgorrie) [inline build/README.md]: This is a very strange readme?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+node_modules
+.DS_Store
+build/*
```
Comment (jamesgorrie) [inline .gitignore]: I've personally tried to start using `.` directories for things like this, but looking at it, `/dist` seems to be more standard.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+node_modules
+.DS_Store
```
Comment (jamesgorrie) [inline .gitignore]: This should be set in your global git ignore as it's not project specific.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,8 @@
-# together-science-can
-Site for the Together Science Can campaign.
+# Together Science Can
+
+This is a single-page site for the *Together Science Can* campaign. Unlike most other Wellcome sites, this one is designed as a static site generated from JSON, which means it's very lightweight and extensible.
```
Comment (jamesgorrie) [inline README.md]: > Unlike most other Wellcome sites

Hopefully not for long - is this sentence useful?

File: components/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+This directory contains HTML templates, JS and SCSS for the site components.
```
Comment (jamesgorrie) [inline components/README.md]: Again, strange to have to explain the use of a dir?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    "url": "git+https://github.com/wellcometrust/together-science-can.git"
+  },
+  "keywords": [
+    "wellcome",
+    "togethersciencecan",
+    "static-site",
+    "nodejs"
+  ],
+  "author": "wellcometrust",
+  "license": "ISC",
```
Comment (jamesgorrie) [inline package.json]: Are we sure not MIT?

File: webpack.config.prod.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+config.plugins.push(
+    new webpack.optimize.UglifyJsPlugin({
+        compress: { warnings: false }
+    }),
+    new webpack.DefinePlugin({
+        ENV: { development: false }
+    })
+);
+
+module.exports = config;
```
Comment (jamesgorrie) [inline webpack.config.prod.js]: From a code perspective, it might just be nicer to copy the config with overrides?

File: webpack.config.dev.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        path: path.join(__dirname, 'build'),
+        filename: 'main.js'
+    },
+    devtool: 'cheap-module-source-map', // enable source maps
+    module: {
+        loaders: [
+              {
+                    test: /\.js$/,
+                    exclude: /node_modules/,
+                    loaders: ['babel-loader?presets[]=env', 'eslint-loader?fix=true']
```
Comment (jamesgorrie) [inline webpack.config.dev.js]: `fix=true` 👍

File: index.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+const html = render(content);
+
+fs.writeFile(
+  path.join(__dirname, 'build', OUTPUT_FILE_NAME),
+  html,
+  err => {
+    if (err)  throw err;
+    else      console.log(`Generated ${OUTPUT_FILE_NAME}.`);
+  }
+);
```
Comment (jamesgorrie) [inline index.js]: Why are we doing this ourselves instread of using something pre-built?

Comment (jakubfiala) [inline index.js]: what would you have in mind? something like Jekyll? This is a completely custom site, not sure if it's worth using an opinionated tool – we're basically just putting together a bunch of nunjucks templates

File: build/README.md (hunk 1 above)
Comment (jakubfiala) [inline build/README.md]: all these oneliner readmes are there just so the folders get commited 😜 this means we don't have to mkdir every time

File: .gitignore (hunk 3 above)
Comment (jakubfiala) [inline .gitignore]: sure, will do

File: .gitignore (hunk 2 above)
Comment (jakubfiala) [inline .gitignore]: ok, can call it `dist`

File: README.md (hunk 4 above)
Comment (jakubfiala) [inline README.md]: hmm, yeah it's prone to be outdated. will remove that bit.

File: package.json (hunk 6 above)
Comment (jakubfiala) [inline package.json]: need to check with @alfraser86, this is just the npm default but I'd prefer MIT

**Comment by jakubfiala** [comment]
@jamesgorrie I'm going to ask ACF if there's a document somewhere explaining the project – it's essentially a one-page campaign site, we want some way of easily changing content but keep it really lightweight, so we've decided to go for a simple node renderer app which renders the page based on JSON content.

**Comment by jakubfiala** [comment]
I can also send you the site designs, you'll see that it's a very simple site, no need for anything more complex than nunjucks

**Comment by jakubfiala** [comment]
going to merge this, if you have any more feedback @jamesgorrie lmk

**Comment by jamesgorrie** [comment]
> node renderer app
Why not serve up an HTML file from S3 - no app required?

**Comment by alfraser86** [comment]
@jamesgorrie He could do that and if pushed for time that could be a good fallback. @jakubfiala is on the Mosaic which comes straight after this so I asked Jakub to see whether he could develop something that would give him a head start on doing that. It also offers the possibility of putting some kind of CMS to create the JSON in the future, although that isn't MVP.

**Comment by jakubfiala** [comment]
> Why not serve up an HTML file from S3 - no app required?

That's exactly what we're doing. The app just generates the HTML file which is then put on S3. It doesn't serve the file.

**Comment by jamesgorrie** [comment]
Ah - if the reason is to work on something that's going to be larger though - shouldn't we be using one of the many frameworks available for this?

**Comment by jakubfiala** [comment]
@jamesgorrie any examples of frameworks?

**Comment by jamesgorrie** [comment]
Netlify and Spike seem great seems to be all the rage?
https://www.netlify.com/
https://github.com/static-dev/spike

**Comment by jamesgorrie** [comment]
There's an interesting article about using GraphCMS and those two:
https://graphcms.com/blog/static-site-generation-using-graphcms-spike-and-netlify

**Pull request merged**